packages/mobile/src/screens/profile-screen/ProfileHeader/ViewAllSupportingTile.tsx

```diff
@@ -51,31 +53,29 @@ export const ViewAllSupportingTile = () => {
     'user_id',
     'supporting_count'
   ])
-  const supportingForProfile: SupportingMapForUser =
-    useSelector((state: CommonState) =>
-      getOptimisticSupportingForUser(state, user_id)
-    ) || {}
-  const rankedSupportingIds = Object.keys(supportingForProfile)
-    .sort((k1, k2) => {
-      const amount1BN = stringWeiToBN(
-        supportingForProfile[k1 as unknown as ID].amount
-      )
-      const amount2BN = stringWeiToBN(
-        supportingForProfile[k2 as unknown as ID].amount
-      )
-      return amount1BN.gte(amount2BN) ? -1 : 1
-    })
-    .map((k) => supportingForProfile[k as unknown as ID])
-    .map((s) => s.receiver_id)
-  const rankedSupporting = useSelector((state) => {
-    const usersMap = getUsers(state, { ids: rankedSupportingIds })
-    return rankedSupportingIds.map((id) => usersMap[id]).filter(Boolean)
-  })
+
+  const rankedSupportingList = useRankedSupportingForUser(user_id)
+
+  const rankedSupportingUsers = useProxySelector(
+    (state) => {
+      const rankedIds = rankedSupportingList.map((s) => s.receiver_id)
+      const usersMap = getUsers(state, {
+        ids: rankedIds
+      })
+      return rankedIds.map((id) => usersMap[id]).filter(Boolean)
+    },
+    [rankedSupportingList]
+  )
 
   const handlePress = useCallback(() => {
     navigation.push('SupportingUsers', { userId: user_id })
   }, [navigation, user_id])
 
+  const viewAllString = useMemo(
```

Comment:
i think this can just be a string since string equality is sufficient

Reply:
Oh yeah, you're right. I think it bothered me to do the formatting logic on every render since the number doesn't change often. But this component shouldn't re-render unless the supporing list changes anyway, so that's probably not a useful optimization.